plugins/webp-uploads/hooks.php

```diff
@@ -171,15 +179,23 @@ function webp_uploads_create_sources_property( array $metadata, int $attachment_
 	}
 
 	// Make sure we have some sizes to work with, otherwise avoid any work.
-	if ( empty( $metadata['sizes'] ) || ! is_array( $metadata['sizes'] ) ) {
+	if (
+		! isset( $metadata['sizes'] ) ||
+		! is_array( $metadata['sizes'] ) ||
+		0 === count( $metadata['sizes'] )
+	) {
 		return $metadata;
 	}
 
 	$sizes_with_mime_type_support = webp_uploads_get_image_sizes_additional_mime_type_support();
 
 	foreach ( $metadata['sizes'] as $size_name => $properties ) {
 		// Do nothing if this image size is not an array or is not allowed to have additional mime types.
-		if ( ! is_array( $properties ) || empty( $sizes_with_mime_type_support[ $size_name ] ) ) {
+		if (
+			! is_array( $properties ) ||
+			! isset( $sizes_with_mime_type_support[ $size_name ] ) ||
+			false === (bool) $sizes_with_mime_type_support[ $size_name ]
```

Comment:
The `$sizes_with_mime_type_support` array is already typed as `array<string, bool>` so no type casting is needed:

```suggestion
			false === $sizes_with_mime_type_support[ $size_name ]
```